airbyte-integrations/bases/base-normalization/dbt-project-template/macros/cross_db_utils/array.sql

```diff
@@ -99,14 +99,22 @@
 
 {% macro default__unnest_cte(from_table, stream_name, column_col) -%}{%- endmacro %}
 
+{%- macro redshift__array_length(column) -%}
+    {%- if redshift_super_type() -%}
+        get_array_length({{ column }})
+    {%- else -%}
+        json_array_length({{ column }}, true)
+    {%- endif -%}
+{%- endmacro -%}
+
 {# -- based on https://blog.getdbt.com/how-to-unnest-arrays-in-redshift/ #}
 {% macro redshift__unnest_cte(from_table, stream_name, column_col) -%}
```

Comment:
As pointed out in the linked article used as a reference for this macro:
> Unnesting arrays on Redshift is a pain, but it’s doable.
https://blog.getdbt.com/how-to-unnest-arrays-in-redshift/

But, with the introduction of the `super` data type, Redshift is in theory more adapted to the JSON data types (which is the point of changing all the types for that), so there should be now better ways for manipulating JSON columns, especially unnesting:
https://bambooengineering.io/unnest-json-objects-in-redshift/

Can we have a closer look to see if we could remove that ugly/inefficient unnesting macro for redshift and use the new native JSON functions?
Or can we document why the `super` data type and new JSON functions are not enough to handle unnesting if it's still not the case?

Reply:
1. About changing option SUPER -> VARCHAR, VARCHAR -> SUPER.
Yes, it works incorrect way for now
we need or update column type, or reset data 
I will discuss with @alexandertsukanov this problem

2. About supporting 2 types or only one SUPER type.
Let's discuss about this more and decide what way we go,
because it affects what we do next to finish this task

3. About ARRAY UNNESTing.
Yes good catch, thanks!
for SUPER type we can use much easier implementation for array UNNESTing
I have re-implemented the more appropriate way.